### PR TITLE
Update INSTALL.md

### PR DESCRIPTION
**Pull request by mdouze**
Added doc + placeholders on how to compile demos and tests with cmake

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -310,6 +304,8 @@ libfaiss.so (or libfaiss.dylib)
 the executable should be linked to one of these. If you use
 the static version (.a), add the LDFLAGS used in the Makefile.
 
+** is this up-to-date? Does cmake generate dynamic libs? **
```
Comment (beauby) [inline INSTALL.md:307]: CMake generate shared libs iff the `-DBUILD_SHARED_LIBS=ON` option is passed to the `cmake` invocation.

File: INSTALL.md [hunk 2]
```diff
@@ -99,35 +100,28 @@ thus does not need an include path.
...
 -------------
 
 A basic usage example is in
 
-  `demos/demo_ivfpq_indexing`
+  `demos/demo_ivfpq_indexing.cpp`
 
-which you can build by calling
-  `make demo_ivfpq_indexing`
+which you can build by calling
```
Comment (beauby) [inline INSTALL.md:114]: Looks like the command disappeared.

Comment (beauby) [reply INSTALL.md:114]: build: `$ make -C build demo_ivfpq_indexing`
run: `$ ./build/demo_ivfpq_indexing`

File: INSTALL.md [hunk 3]
```diff
@@ -99,35 +100,28 @@ thus does not need an include path.
...
+** how to run cmake in the demos subdirectory **
 
 It makes a small index, stores it and performs some searches. A normal
 runtime is around 20s. With a fast machine and Intel MKL's BLAS it
 runs in 2.5s.
 
 To run the whole test suite:
 
-   `make test` (for the CPU part)
+** how to run the C++ tests **
```
Comment (beauby) [inline INSTALL.md:124]: `$ make -C build test`

File: INSTALL.md [hunk 4]
```diff
@@ -143,10 +137,7 @@ directory for this repository.
 
 Then compile and run the following (after ensuring you have installed faiss):
 
-```
-make demos
-./demos/demo_sift1M
-```
+** how to compile demo_sift1M **
```
Comment (beauby) [inline INSTALL.md:140]: `$ make -C build demo_sift1M`

File: INSTALL.md [hunk 5]
```diff
@@ -159,8 +150,9 @@ Python Interface
 The Python interface is provided via SWIG (Simple Wrapper and
 Interface Generator) and an additional level of manual wrappers (in faiss/python/faiss.py).
 
-SWIG generates two wrapper files: a Python file (`faiss/python/swigfaiss.py`) and a
-C++ file that must be compiled to a dynamic library (`faiss/python/_swigfaiss.so`).
+SWIG generates two wrapper files: a Python file `swigfaiss.py` and a
+C++ file that must be compiled to a dynamic library (`_swigfaiss.so`).
+There is an AVX2 variant of the files, suffixed with `_avx2`.
```
Comment (beauby) [inline INSTALL.md:155]: This part is not really relevant to installing, and is probably more confusing than helpful to new users.

File: INSTALL.md [hunk 6]
```diff
@@ -231,13 +223,14 @@ nvcc, except some of them that are not recognized and that should be
...
 
-You may want to add `-j 10` to use 10 threads during compile.
+You may want to add `-j 10` to the `make` command to use 10 threads during compile.
 
 Testing the GPU implementation
 ------------------------------
 
 Compile the example with
 
+** update **
```
Comment (beauby) [inline INSTALL.md:233]: `$ make -C build demo_ivfpq_indexing_gpu`

File: INSTALL.md [hunk 7]
```diff
@@ -258,6 +251,7 @@ use_gpu = True
 
 and you can run
 
+** update ** 
 ```
 export PYTHONPATH=.
 python demos/demo_auto_tune.py
```
Comment (beauby) [inline INSTALL.md:257]: At the moment, running the python tests involves the following:
```
$ make -C build swigfaiss
$ (cd build/faiss/python && setup.py build)
$ PYTHONPATH=./build/faiss/python/build/lib python -m unittest discover ./tests
```

File: INSTALL.md (hunk 3 above)
Comment (mdouze) [reply INSTALL.md:124]: Does not work
```
(faiss_latest_cmake) matthijs@devfair0144:~/faiss_versions/faiss_latest_cmake$ make -C build test
make: Entering directory '/private/home/matthijs/faiss_versions/faiss_latest_cmake/build'
make: *** No rule to make target 'test'.  Stop.
make: Leaving directory '/private/home/matthijs/faiss_versions/faiss_latest_cmake/build'
(faiss_latest_cmake) matthijs@devfair0144:~/faiss_versions/faiss_latest_cmake$ make -C build tests
make: Entering directory '/private/home/matthijs/faiss_versions/faiss_latest_cmake/build'
make: *** No rule to make target 'tests'.  Stop.
make: Leaving directory '/private/home/matthijs/faiss_versions/faiss_latest_cmake/build'
```

File: INSTALL.md (hunk 4 above)
Comment (mdouze) [reply INSTALL.md:140]: Does not work due to MKL linking

https://gist.github.com/mdouze/975b681c1a45f03664851d646a6f4243

File: INSTALL.md (hunk 5 above)
Comment (mdouze) [reply INSTALL.md:155]: Will move it to the packaging section.

File: INSTALL.md (hunk 3 above)
Comment (beauby) [reply INSTALL.md:124]: In order to build the tests, they need to be enabled via `-DBUILD_TESTING=ON`. The option should be `ON` by default, but it is currently not, I'll send a fix.

File: INSTALL.md (hunk 4 above)
Comment (beauby) [reply INSTALL.md:140]: `$ cmake -B build -DBLA_VENDOR=Intel10_64_dyn -DFAISS_ENABLE_GPU=OFF .`
`$ make -C build -j demo_sift1M`
works fine for me. There may be an issue with your MKL installation?

**Pull request closed without merging**